### ISSUE TITLE
Thread-safety for __start arguments parsing in crt0.h

### ISSUE THREAD

**Issue by rben-dev** [picolibc/picolibc#735]
Hi,

Following some tests with `picolibc` and `qemu` semi-hosting on Risc-v, I stumbled across a possible issue with command line arguments parsing from https://github.com/picolibc/picolibc/blob/695583c3b7dc461204948db5c8390e967b526a27/picocrt/crt0.h#L106: the `static` local variables for the arguments do not seem to be thread-safe. This yields erratic behaviors when `__start` is executed on multiple cores, which seems to be the case for the Risc-v `sifive_u` board for instance. See the discussion on the `qemu` repository where I opened an issue a few days ago: https://gitlab.com/qemu-project/qemu/-/issues/2332.

Although my knowledge is limited regarding the expected behavior of such a `-kernel` execution (i.e. should `qemu` spinloop the other cores or perform parallel execution as it is done now, when compared to what a real Risc-v multi-cores hardware board would do), I find that `picolibc`   `__start` could be rendered thread-safe by using non `static` variables or by other means: any thoughts on this?

Thanks in advance,
Regards,

**Comment by keith-packard** [comment]
Yeah, these sample startup scripts are designed to show what the library needs from the underlying system as well as providing a mechanism for running the builtin picolibc tests. Multi-core systems aren't really a design target. In this particular case, making the variables thread-local wouldn't be sufficient as there isn't any per-thread TLS setup provided in the architecture-specific code. We'd have to also allocate per-core stacks somewhere. If you think that would be fun to play with, I'd be happy to review the code and see if we could integrate that into these samples though.
